arangod/RestHandler/RestDocumentHandler.cpp

```diff
@@ -677,21 +681,20 @@ RestStatus RestDocumentHandler::readManyDocuments() {
     return RestStatus::DONE;
   }
 
-  return waitForFuture(_activeTrx->documentAsync(cname, search, opOptions)
-                       .thenValue([=](OperationResult opRes) {
-    auto res = _activeTrx->finish(opRes.result);
-
-    if (opRes.fail()) {
-      generateTransactionError(cname, opRes);
-      return;
-    }
+  return waitForFuture(_activeTrx->documentAsync(cname, search, opOptions).thenValue([=](OperationResult opRes) {
+    return _activeTrx->finishAsync(opRes.result).thenValue([=, opRes(std::move(opRes))](Result&& res) {
```

Comment:
Same here. If this is just a local operation, there is no need to make it async. If it does communication with followers, it is necessary.